pkg/ls/compile.go

```diff
@@ -23,7 +23,7 @@ import (
 // A CompiledGraph is a graph of compiled schemas
 type CompiledGraph interface {
 	GetCompiledSchema(string) *Layer
-	PutCompiledSchema(string, *Layer) *Layer
+	PutCompiledSchema(string, *Layer, Context) *Layer
```

Comment:
Make Context the first arg to functions. That's a well established convention.

Pass *Context, not Context